v2/client/client.go

```diff
@@ -110,3 +99,23 @@ func (c *Client) doVoucherRequest(ctx context.Context, url string, image referen
 	}
 	return &voucherResp, nil
 }
+
+func newClient(voucherURL string, httpClient *http.Client) (*Client, error) {
+	if voucherURL == "" {
+		return nil, errNoHost
+	}
+
+	u, err := url.Parse(voucherURL)
+	if nil != err {
```

Comment:
nit pick: can we either use `err !=nil`  or `nil != err`?  I personally think `err != nil` is more readable. Either option is fine, as long as we use the same style throughout the code? (At least in `client.go`)?